src/Utilities/Compiler/Lightup/SemanticModelExtensions.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis;
+
+namespace Analyzer.Utilities.Lightup
+{
+    internal static class SemanticModelExtensions
+    {
+        private static readonly Func<SemanticModel, int, NullableContext> s_getNullableContext
+            = LightupHelpers.CreateAccessorWithArgument<SemanticModel, int, NullableContext>(typeof(SemanticModel), "semanticModel", typeof(int), "position", nameof(GetNullableContext));
+
+        public static NullableContext GetNullableContext(this SemanticModel semanticModel, int position)
+            => s_getNullableContext(semanticModel, position);
+    }
+}
```

Comment:
Although I understand this whole set of helpers was initially devised for another PR and will be indirectly tested, it would probably be good to add a few tests in this PR to cover.

Reply:
I added a few tests for a subset of the core helpers. Covering these cases in tests is hard.